src/app/page.tsx

```diff
@@ -64,88 +54,38 @@ export default function Home() {
 }
 
 const SonarEntityPanel = () => {
-  const client = useSonarClient();
-  const { authenticated, ready } = useSonarAuth();
-  const { address, isConnected } = useAccount();
-
-  const [entity, setEntity] = useState<ResourceState<EntityDetails>>({
-    hasFetched: false,
-    loading: false,
-    value: undefined,
+  const { address: walletAddress, isConnected } = useAccount();
+  const { authenticated, loading, entity, error } = useSonarEntity({
+    saleUUID: sonarConfig.saleUUID,
+    wallet: { address: walletAddress, isConnected },
```

Comment:
having to pass `isConnected` is slightly sus. why not just `walletAddress: undefined`? I guess you've already discussed this at length with Willyham?

Reply:
No not too much thought went into this to be honest. It's really just because those are return fields that the wagmi useAccount provides. Reading the docs now though I think `!isConnected` implies `address == undefined`, so yeah I think isConnected is redundant here.

I'll remove in the library and then update this/the docs.

Reply:
https://github.com/sunrisedotdev/sonar/pull/8